src/backends/onnxruntime.c

```diff
@@ -348,28 +415,41 @@ int RAI_ModelRunORT(RAI_ModelRunCtx* mctx, RAI_Error *error) {
     const char* msg = OrtGetErrorMessage(status);
     RAI_SetError(error, RAI_EMODELRUN, msg);
     OrtReleaseStatus(status);
+    OrtReleaseAllocator(allocator);
     return 1;
   }
 
   for (size_t i=0; i<n_output_nodes; i++) {
-    RAI_Tensor* output_tensor = RAI_TensorCreateFromOrtValue(outputs[i]);
-    mctx->outputs[i].tensor = RAI_TensorGetShallowCopy(output_tensor);
-    RAI_TensorFree(output_tensor);
- 
+    RAI_Tensor* output_tensor = RAI_TensorCreateFromOrtValue(outputs[i], error);
+    if (error->code != RAI_OK) {
+      OrtReleaseStatus(status);
+      OrtReleaseAllocator(allocator);
+      return 1;
+    }
+    if (output_tensor) {
+      mctx->outputs[i].tensor = RAI_TensorGetShallowCopy(output_tensor);
+      RAI_TensorFree(output_tensor);
+    }
+    else {
+      printf("ERR: non-tensor output from ONNX models, ignoring (currently unsupported).\n");
+    }
     OrtReleaseValue(outputs[i]);
   }
 
   for (size_t i=0; i<n_input_nodes; i++) {
     OrtReleaseValue(inputs[i]);
   }
 
+  OrtReleaseAllocator(allocator);
```

Comment:
Need to call OrtReleaseStatus() is successful?

Reply:
In the ONNXRuntime API, `status` is only ever non-NULL if there's an issue with the function returning `status`. So if everything goes well, `status` is never allocated.